src/Razor/Microsoft.AspNetCore.Razor.Language/src/RazorProjectEngine.cs

```diff
@@ -226,7 +226,8 @@ private static void AddComponentFeatures(RazorProjectEngineBuilder builder, Razo
             // into the repository.
             ComponentTypeParamDirective.Register(
                 builder,
-                supportConstraints: razorLanguageVersion.CompareTo(RazorLanguageVersion.Version_6_0) >= 0);
+                supportConstraints: true);
+                //supportConstraints: razorLanguageVersion.CompareTo(RazorLanguageVersion.Version_6_0) >= 0);
```

Comment:
Perhaps it might be made better with an explainer comment but the intent of this is to show what the actual correct check should be since we shouldn't be enabling this feature unconditionally.